build.gradle.kts

```diff
@@ -93,6 +93,9 @@ kotlin {
                 implementation("io.kotest:kotest-assertions-core:6.0.0.M1")
                 implementation("io.kotest:kotest-framework-api:6.0.0.M1")
                 implementation("io.kotest:kotest-framework-engine:6.0.0.M1")
+                // Overridig coroutines' version to solve a problem with WASM JS tests.
+                // See https://kotlinlang.slack.com/archives/CDFP59223/p1736191408326039?thread_ts=1734964013.996149&cid=CDFP59223
+                runtimeOnly("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.10.1")
```

Comment:
We have this workaround in snakeyaml-engine-kmp as well:
https://github.com/krzema12/snakeyaml-engine-kmp/blob/a447f091a9f27639dbb3f4b2bd11d49adf411329/build.gradle.kts#L43_l46

Reply:
We'll be able to remove this workaround once the next release of kotest appears (see https://github.com/kotest/kotest/issues/4644).

Reply:
Is there a snapshot or pre-release version of Kotest we could use instead?

Reply:
We're already on a prerelease (6.0.0.M1), and there are snapshots according to https://kotest.io/docs/quickstart/#snapshots there are snapshots published from the main branch. I can test it out if that's what you prefer, although it may be even less stable and hacky.

Reply:
The snapshots seem to be broken, here https://oss.sonatype.org/content/repositories/snapshots/io/kotest/kotest-assertions-core-jvm/ I see versions up to `5.6.0.1131-SNAPSHOT`. Asked here: https://kotlinlang.slack.com/archives/CT0G9SD7Z/p1737962946077139

Reply:
Tried with the snapshot, but the tests fail on CI (they pass locally): https://github.com/charleskorn/kaml/actions/runs/12986349503/job/36212971272?pr=664

So we either wait for a proper release of kotest with the coroutines updated, or use the workaround I proposed earlier.

Reply:
The workaround from earlier sounds good to me.

Reply:
Ok, reverted to the original workaround! Once this PR is merged, other PRs are likely good to be merged as well.